server/chat-commands/info.ts

```diff
@@ -2569,10 +2572,16 @@ export const commands: ChatCommands = {
 			const [width, height, resized] = request.dimensions;
 			buf = Utils.html`<img src="${request.link}" width="${width}" height="${height}" />`;
 			if (resized) buf += Utils.html`<br /><a href="${request.link}" target="_blank">full-size image</a>`;
-		} else {
-			const YouTube = new YoutubeInterface();
-			buf = await YouTube.generateVideoDisplay(request.link);
-			if (!buf) return this.errorReply('Could not get YouTube video');
+		}
+		const Spotify = new SpotifyInterface();
+		const YouTube = new YoutubeInterface();
+		if (Spotify.linkRegex.test(request.link) || YouTube.linkRegex.test(request.link)) {
+			if (Spotify.linkRegex.test(request.link)) {
+				buf = Spotify.generateSongDisplay(request.link);
+			} else {
+				buf = await YouTube.generateVideoDisplay(request.link);
+				if (!buf) return this.errorReply('Could not get YouTube video');
+			}
```

Comment:
```suggestion
		if (Spotify.linkRegex.test(request.link)) {
			buf = Spotify.generateSongDisplay(request.link);
		} else if (YouTube.linkRegex.test(request.link)) {
			buf = await YouTube.generateVideoDisplay(request.link);
			if (!buf) return this.errorReply('Could not get YouTube video');
		}
```

Reply:
Do we need an error for being unable to fetch a Spotify song?

Reply:
No, this doesn't use any sort of fetcher, just a tag.

Reply:
Not sure about that.